fineract-provider/src/main/java/org/apache/fineract/infrastructure/s3/AmazonS3ConfigCondition.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.s3;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.providers.AwsRegionProvider;
+
+public class AmazonS3ConfigCondition implements Condition {
+
+    @Value("${fineract.report.export.s3.enabled}")
+    private boolean amazonS3Enabled;
+
+    @Value("${fineract.report.export.s3.bucket:reports}")
+    private String bucketName;
+
+    @Override
+    public boolean matches(ConditionContext context, AnnotatedTypeMetadata metadata) {
+        return context.getEnvironment().getRequiredProperty("fineract.report.export.s3.enabled", Boolean.class)
+                && StringUtils.isNotBlank(context.getEnvironment().getProperty("fineract.report.export.s3.bucket", ""))
+                && isAwsCredentialValid(context);
+    }
+
+    private boolean isAwsCredentialValid(ConditionContext context) {
+        try {
+            context.getBeanFactory().getBean(AwsCredentialsProvider.class).resolveCredentials();
```

Comment:
This is not necessarily the best idea. You're accessing the container before it even gets properly initialized. 
It's better if you explicitly instantiate the default credentials provider and region provider to test this.

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Condition.html

> Conditions must follow the same restrictions as [BeanFactoryPostProcessor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanFactoryPostProcessor.html) and take care to never interact with bean instances.